### PR TITLE
release: ci/cd release update (PROJQUAY-1486)

### PR DESCRIPTION
**Pull request by thomasmckay**
PROJQUAY-1486

* Update pull-request template and commit message check
* Add first version of clair-style release github actions (follow-up PR to finalize)

Commit message must be of the form
```
<scope>: <subject> (PROJQUAY-1234)
<optional details>
```

### CODE REVIEW

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -178,3 +178,19 @@ jobs:
...
+
+  commit-check:
+    name: Commit Check
+    runs-on: ubuntu-latest
+    steps:
+      - name: commit check
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: |
+            ^(.*):\s*(.*)\nPROJQUAY.*$
```
Comment (kleesc) [inline .github/workflows/CI.yaml:190]: I'd say if we're going to enforce that, put the JIRA ticket on the same line as the header.
i.e. `^(.*):\s*(.*)\s\(PROJQUAY-[0-9]+\)$`

**Review by kleesc — CHANGES_REQUESTED** [review]

File: .github/workflows/CI.yaml (hunk 1 above)
Comment (thomasmckay) [reply .github/workflows/CI.yaml:190]: i like that idea

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,24 +1,3 @@
-**Issue:** https://issues.redhat.com/browse/PROJQUAY-???
-Pull-request title must start with "PROJQUAY-??? - "
+Please follow commit message requirements:
+<scope>: <subject> (PROJQUAY-####)
```
Comment (kleesc) [inline .github/PULL_REQUEST_TEMPLATE.md:2]: We don't really need this template if we're going to enforce a commit message format.
This should be in a README or HOWTO for contributing or something.

File: .github/workflows/CI.yaml [hunk 3]
```diff
@@ -178,3 +178,19 @@ jobs:
...
+
+  commit-check:
+    name: Commit Check
+    runs-on: ubuntu-latest
+    steps:
+      - name: commit check
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: |
+            ^(.*):\s*(.*)\s\(PROJQUAY-[0-9]+\).*$
```
Comment (kleesc) [inline .github/workflows/CI.yaml:190]: Add a new line before the end: `^(.*):\s*(.*)\s(\(PROJQUAY-[0-9]+\))\n.*$`

**Review by kleesc — CHANGES_REQUESTED** [review]

File: .github/workflows/CI.yaml (hunk 3 above)
Comment (thomasmckay) [reply .github/workflows/CI.yaml:190]: i had that originally but it was failing my one line commit message, are you saying it's required for the actions?

Comment (kleesc) [reply .github/workflows/CI.yaml:190]: As it is now `scope: somemessage header (PROJQUAY-1234 )and some other stuff` would be valid.
New line makes sure the last part of the header is the ticket number.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (thomasmckay) [reply .github/PULL_REQUEST_TEMPLATE.md]: removed

File: .github/workflows/CI.yaml (hunk 3 above)
Comment (thomasmckay) [reply .github/workflows/CI.yaml]: updated but you can see that it is now failing my single line commit

Comment (kleesc) [reply .github/workflows/CI.yaml]: Right, you should add a new line to your message.
It is generally good practice to have a commit message anyways.

Comment (thomasmckay) [reply .github/workflows/CI.yaml]: changed to accept single or multi line

**Review by kleesc — APPROVED** [review]

**Pull request merged**